src/mps/observer.jl

```diff
@@ -31,7 +31,6 @@ struct DMRGObserver <: AbstractObserver
   ops::Vector{String}
   sites::Vector{<:Index}
   measurements::Dict{String,DMRGMeasurement}
-  #energies::Vector{Float64}
   energies::Vector{Complex}
```

Comment:
I think it would be nice to make it parametrized like this:
```julia
struct DMRGObserver{T} <: AbstractObserver
  ops::Vector{String}
  sites::Vector{<:Index}
  measurements::Dict{String,DMRGMeasurement}
  energies::Vector{T}
  truncerrs::Vector{Float64}
  etol::Float64
  minsweeps::Int64
end
```
I don't think that should be an intrusive change, and would keep the `energies` as a concrete type.

Reply:
In general you should not have to write `DMRGObserver{Float64}`. The default Julia constructor should infer the type, and if not we can write a constructor for `DMRGObserver` that infers the type parameter.

Reply:
I believe the typical way of making a DMRGObserver, though, is to either just default-construct it or pass a set of operator names to measure, neither of which specifies the energy type. So one way to go is to default to Float64 and then have users have to specify ComplexF64 if that's what they want. Another is to always require it be specified. 

(Of course a lot of this goes to show how the new Observers.jl is nicer!)